### PR TITLE
Add NVIDIA Nsight Systems and Nsight Compute documentation

### LINKED ISSUE #174: Nsight Compute should be documented as in the NVIDIA profiler section
Nsight Compute now works on IBM Power systems as of CUDA 10.1 Update 1, there should be a description of how to use it for kernel-level profiling in the profilers section.

Related to #60.

### PR DESCRIPTION
**Pull request by maxpkatz**
Closes #174

### CODE REVIEW

File: systems/summit_user_guide.rst [hunk 1]
```diff
@@ -3030,15 +3030,171 @@ Page <https://www.olcf.ornl.gov/software_package/valgrind/>`__.
...
+
+The most important output to look at is the "GPU Speed of Light" section,
+which tells you what fraction of peak memory throughput and what fraction
+of peak compute throughput you achieved. Typically if you have achieved
+higher than 60% of the peak of either subsystem, your kernel would be
+considered memory-bound or compute-bound (respectively), and if you have
+not achieved 60% of either this is often a latency-bound kernel. (A common
+cause of latency issues is not exposing enough parallelism to saturate
+the GPU's compute capacity -- peak GPU performance can only be achieved
+when there is enough work to hide the relatively long latency of memory
```
Comment (fs-nv) [inline systems/summit_user_guide.rst:3143]: I would phrase slightly differently: "when there is enough work to hide the latency of memory access and to keep all compute pipelines busy."

File: systems/summit_user_guide.rst [hunk 2]
```diff
@@ -3030,15 +3030,171 @@ Page <https://www.olcf.ornl.gov/software_package/valgrind/>`__.
...
+accesses on the GPU).
+
+By default, Nsight Compute will collect this performance data for every
+kernel in your application. This will take a long time in a real application.
```
Comment (fs-nv) [inline systems/summit_user_guide.rst:3147]: maybe "real" -> "real-world"

File: systems/summit_user_guide.rst [hunk 3]
```diff
@@ -3030,15 +3030,171 @@ Page <https://www.olcf.ornl.gov/software_package/valgrind/>`__.
...
+
+If you want to collect information on just a specific performance measurement,
+for example the number of bytes written to DRAM, you can do so with the
+``--metrics`` option:
+
+::
+
+    summit> jsrun -n1 -a1 -g1 nv-nsight-cu-cli -k vectorAdd --metrics dram__bytes_write.sum ./vectorAdd
+
+The list of available metrics can be obtained with ``nv-nsight-cu-cli --query-metrics``.
```
Comment (fs-nv) [inline systems/summit_user_guide.rst:3163]: Might be worth pointing out that those metric names need to be added a suffix to be collectible, which is a common cause of failure.

File: systems/summit_user_guide.rst [hunk 4]
```diff
@@ -3030,15 +3030,171 @@ Page <https://www.olcf.ornl.gov/software_package/valgrind/>`__.
...
+application. If that performance bottleneck is on the CPU, it means more
+code should be ported to the GPU; or, if that bottleneck is in memory
+management, such as copying data back and forth between the CPU and GPU,
+you should look for opportunities to reduce that data motion. But if that
+bottleneck is a GPU kernel, then Nsight Compute can be used to collect
+performance counters to understand whether the kernel is running efficiently
+and if there's anything you can do to improve.
+
+The Nsight Compute command-line interface, ``nv-nsight-cu-cli``, can be
+prefixed to your application to collect a report.
```
Comment (fs-nv) [inline systems/summit_user_guide.rst:3125]: You might want to point out that Nsight Compute also creates a temporary report file, even when -o is not specified, just like Nsight Systems.

File: systems/summit_user_guide.rst [hunk 5]
```diff
@@ -3030,15 +3030,171 @@ Page <https://www.olcf.ornl.gov/software_package/valgrind/>`__.
...
+``--metrics`` option:
+
+::
+
+    summit> jsrun -n1 -a1 -g1 nv-nsight-cu-cli -k vectorAdd --metrics dram__bytes_write.sum ./vectorAdd
+
+The list of available metrics can be obtained with ``nv-nsight-cu-cli --query-metrics``.
+
+As with Nsight Systems, there is a graphical user interface you can load
+a report file into. Use the ``-o`` flag to create a file (the added report
```
Comment (fs-nv) [inline systems/summit_user_guide.rst:3166]: This graphical user interface is also not available on Power, but on Windows, x86_64 Linux and Mac.

File: systems/summit_user_guide.rst (hunk 1 above)
Comment (jack-morrison) [reply systems/summit_user_guide.rst]: How does 2603922 look?

File: systems/summit_user_guide.rst (hunk 3 above)
Comment (jack-morrison) [reply systems/summit_user_guide.rst]: Are the changes in 11d5a90 correct? I'm not terribly familiar with NSight {Compute, Systems} but referenced the Nvidia docs for this one.

Comment (fs-nv) [reply systems/summit_user_guide.rst]: That's correct.

File: systems/summit_user_guide.rst (hunk 1 above)
Comment (fs-nv) [reply systems/summit_user_guide.rst]: I would use "hide the latency of memory accesses" (i.e. plural), otherwise, looks correct.

Comment (jack-morrison) [reply systems/summit_user_guide.rst]: Fixed in 02375fa

**Comment by jack-morrison** [comment]
@maxpkatz anything else before I merge this up?

**Comment by jack-morrison** [comment]
Considering this 👍 ahead of the hackathon next week. Let's take up any addt'l changes in separate PRs.

Thank you @maxpkatz and @fs-nv for this!

**Comment by maxpkatz** [comment]
Thanks @jack-morrison for finishing it up and merging.

**Pull request merged**